snuba/datasets/entities/transactions.py

```diff
@@ -85,21 +98,119 @@
     ],
 )
 
+SAMPLE_RATE = 0.1
+
+
+class SampledSimplePipelineBuilder(SimplePipelineBuilder):
+    def build_planner(
+        self, query: LogicalQuery, settings: RequestSettings,
+    ) -> EntityQueryPlanner:
+        new_query = deepcopy(query)
+        new_query.set_sample(SAMPLE_RATE)
+        return super().build_planner(new_query, settings)
+
 
 class BaseTransactionsEntity(Entity, ABC):
     def __init__(self, custom_mappers: Optional[TranslationMappers] = None) -> None:
         storage = get_writable_storage(StorageKey.TRANSACTIONS)
         schema = storage.get_table_writer().get_schema()
 
+        pipeline_builder = SimplePipelineBuilder(
+            query_plan_builder=SingleStorageQueryPlanBuilder(
+                storage=storage,
+                mappers=transaction_translator
+                if custom_mappers is None
+                else transaction_translator.concat(custom_mappers),
+            ),
+        )
```

Comment:
you use this variable only at the very end of this method, if you moved the declaration there the code would be more readable.

Reply:
I restructured things a bit.